src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,99 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int ARRAY_DOUBLING = 2;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(getHash(key), key, value, null);
+        } else {
+            Node<K, V> currentNode = table[index];
+            while (currentNode != null) {
+                if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                    currentNode.value = value;
+                    return;
+                }
+                if (currentNode.next == null) {
+                    currentNode.next = new Node<>(getHash(key), key, value, null);
+                    break;
+                }
+                currentNode = currentNode.next;
+            }
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (currentNode.key == key || key != null && key.equals(currentNode.key)) {
+                return currentNode.value;
+            }
+            currentNode = currentNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        Node<K, V>[] oldTable;
+        oldTable = table;
+        table = new Node[table.length * ARRAY_DOUBLING];
+        threshold *= ARRAY_DOUBLING;
+        for (Node<K, V> currentNode : oldTable) {
+            if (currentNode != null) {
+                Node<K, V> oldNode = currentNode;
```

Comment:
You don't need this check and you don't need the variable if currentNode cannot be null